src/main/java/com/deque/html/axecore/selenium/AxeBuilder.java

```diff
@@ -0,0 +1,442 @@
+   * Note that the selectors array uniquely identifies one element in the page.
+   * Refer include(string[]) for more information on the usage
+   * @param selectors Any valid CSS selectors
+   * @return an Axe Builder
+   */
+  public AxeBuilder exclude(final List<String> selectors) {
+    validateParameters(selectors);
+    if (selectors.isEmpty()) {
+      return this;
+    }
+    runContext.setExclude(selectors);
+    return this;
+  }
+
+  /**
+   * Causes analyze() to write the axe results as a JSON file,
+   * in addition to returning it in object format as usual.
+   * @param path Path to the output file.
+   *             Will be passed as-is to the System.IO APIs.
+   * @return an Axe builder
+   */
+  public AxeBuilder withOutputFile(final String path) {
+    validateNotNullParameter(path);
+    outputFilePath = path;
+    return this;
+  }
+
+  /**
+   * Run axe against a specific WebElement
+   * or webElements (including its descendants).
+   * @param context WebElement(s) to test
+   * @return An axe results document
+   */
+  public Results analyze(final WebDriver webDriver, final WebElement... context) {
+    return analyzeRawContext(webDriver, context, true);
+  }
+
+  /**
+   * Run axe against the entire page.
+   * @return An axe results document
+   */
+  public Results analyze(final WebDriver webDriver) {
+    boolean runContextHasData = this.runContext.getInclude() != null
+        || this.runContext.getExclude() != null;
+    String rawContext = runContextHasData
+        ? AxeReporter.serialize(runContext) : null;
+    return analyzeRawContext(webDriver, rawContext, true);
+  }
+
+  /**
+   * Run axe against the entire page.
+   * @return An axe results document
+   */
+  public Results analyze(final WebDriver webDriver, boolean injectAxe) {
+    boolean runContextHasData = this.runContext.getInclude() != null
+        || this.runContext.getExclude() != null;
+    String rawContext = runContextHasData
+        ? AxeReporter.serialize(runContext) : null;
+    return analyzeRawContext(webDriver, rawContext, injectAxe);
+  }
+
+
+  /**
+   * Runs axe via axeRunScript at a specific context, which will be passed
+   * as-is to Selenium for scan.js to interpret, and parses/handles
+   * the scan.js output per the current builder options.
+   * @param rawContextArg The value to pass as-is to
+   *                      scan.js to use as the axe.run "context" argument
+   * @return an Axe Result
+   */
+  private Results analyzeRawContext(final WebDriver webDriver, final Object rawContextArg, boolean injectAxe) {
+    validateNotNullParameter(webDriver);
+    String rawOptionsArg = getOptions().equals("{}")
+        ? AxeReporter.serialize(runOptions) : getOptions();
+    Object[] rawArgs = new Object[] {rawContextArg, rawOptionsArg};
+
+    if (injectAxe) {
+      try {
+        WebDriverInjectorExtensions.inject(
+            webDriver, builderOptions.getScriptProvider());
+      } catch (IOException e) {

```

Comment:
Why catch two different exceptions but do the same thing for both?

Reply:
Good point. Even if an unexpected error occurs we probably still want to tell the user that it was a problem with injection.